### ISSUE TITLE
Google AdWords plugin on PHP7.3

### ISSUE THREAD

**Issue by havenswift-hosting** [cubecart/v6#2394]
Ref : https://github.com/cubecart/v6/issues/2282 - one more instance of error in CubeCart error table

[<strong>Deprecated</strong>] /home/XXXXXXXX/public_html/modules/plugins/Google_Adwords/hooks/class.gui.javascripts.php:37 - Function create_function() is deprecated

**Comment by abrookbanks** [comment]
Fixed with 2.0.2.

**Comment by nathanbright** [comment]
This fix has broken the conversion event because function() (used to replace the deprecated create_function()) is not re-entrant. The second call to registerFilter() is overwriting the first, thus preventing the GA_GLOBAL_CONVERSION_JS script from being included - but it *is* required by the conversion event

One solution is an undocumented feature, passing a third parameter "name" to registerFilter(). Being undocumented this is probably not a good idea

Another is to concatenate the global and event JS files before calling registerFilter()

Apologies if this should be raised as a new bug - I'll do that if my comment here appears to be overlooked

**Comment by nathanbright** [comment]
The issue is more complex. We've more than one plugin needing to use this code and only the last one invoked gets rendered

On reflection the [Smarty docs ](https://www.smarty.net/docs/en/advanced.features.outputfilters.tpl) have shown an alternative solution, create separate named functions and pass the function name to the registerFilter() call

I'm attaching a proposed solution for /plugins/Google_Adwords/hooks/class.gui.javascripts.php

[class.gui.javascripts.php.txt](https://github.com/cubecart/v6/files/5424170/class.gui.javascripts.php.txt)

**Comment by abrookbanks** [comment]
<img width="590" alt="Screenshot 2022-12-14 at 16 41 30" src="https://user-images.githubusercontent.com/4520803/207655295-579459d7-a150-4873-9f07-b26476d61e38.png">
Didn't quite work for me.. 

I'll experiment.

**Comment by abrookbanks** [comment]
Finally fixed with 1.1.1.
```
$purchase_event = "<script id=\"ga4\">$1gtag(\"event\", \"purchase\", ".json_encode($purchase).");</script>";
return preg_replace('/<script id="ga4">([\w\W]*?)<\/script>/s', $purchase_event, $tpl_output, 1);
```
